Dashboard/app/js/lib/controllers/permissions.js

```diff
@@ -137,7 +137,56 @@ FLOW.permControl = Ember.Controller.create({
     this.perms.forEach(function (item) {
       //this.set(item.perm,item.value);
     });
-  }
+  },
+
+  /* Given an entity, process the permissions settings for the current user
+    and return the permissions associated with that entity.  Entity is an Ember object*/
+  permissions: function(entity) {
+    var keyId, ancestorIds, permissions = [], currentUserPermissions = FLOW.userControl.currentUserPathPermissions();
```

Comment:
@muloem For my own understanding.. Does `FLOW.userControl.currentUserPathPermissions()` return a map from `id => list of permissions`? For example
```
{
    "123": ["FORM_CREATE", "FORM_READ"],
    "432": ["DATA_CLEANING", "DATA_DELETE"]
}
```